authentication.py

```diff
@@ -1,124 +1,138 @@
 # Josh Aaron Miller 2021
 # Vennt authentication-related processing
 
-import hashlib, time, threading
+import hashlib
+import secrets
+import time
+import threading
 
 TIMEOUT = 3600 * 12
 
 # VenntHandler methods
+
+
 def handle_register(self, json_data):
-	result = {}
-	username = json_data["register"][0]
-	if self.server.db.is_valid("accounts", username):
-		result["success"] = False
-		result["info"] = "Username already exists"
-		self.respond(result)
-		return
-	if "password" in json_data:
-		password = json_data["password"][0]
-	else:
-		result["success"] = False
-		result["info"] = "No password key"
-		self.respond(result)
-		return
-	self.server.db.create_account(username, hashlib.md5(password.encode('utf-8')).hexdigest())
-	result["success"] = True
-	result["info"] = "Account created"
-	auth_token = hashlib.md5((username + str(time.time())).encode('utf-8')).hexdigest()
-	self.server.db.auth.authenticate(self.client_address[0], username, auth_token)
-	result["auth_token"] = auth_token
-	self.respond(result)
-	return
+    result = {}
+    username = json_data["register"]

```

Comment:
DEV NOTE: We need to take the full username / password not just the first character (or this is super insecure lol)
Also, added password salting since we will need to restart the DB anyway.

Reply:
Wait this was taking the first character only? lol
I think when I wrote it, it had to grab the string out of an array. Maybe the syntax changed?

Reply:
Yeah the syntax must have changed then since it's not definitely not currently in an array  😅